### PR TITLE
Category new heating

### PR DESCRIPTION
**Pull request by smilenow24**
PR for Q-Channel

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,109 +1,166 @@
...
+  const [activeChart, setActiveChart] = useState(true);
 
   const totalConsumption = getTotalConsumption(energyConsumptionHistory);
-  const oldInputLength = initialInputData.length;
-  const dailyTotalBudget = 1000;
-  const chartInputData = updateChart();
 
+  const dailyTotalBudget = 1000;
+  const oldInputElectricLength = initialInputData.electric.length;
+  console.log(energyConsumptionHistory.mobility[0].value);
```
Comment (stefankorder) [inline src/App.js]: PLease delete this console log. In your final PR should be no console logs.

File: src/App.js [hunk 2]
```diff
@@ -1,109 +1,166 @@
...
+      if (energyConsumptionHistory.electric[0].increase > dailyTotalBudget) {
         setMessageText(messages[2].text);
       }
     } else {
-      setMessageText('please input > or = ' + energyConsumptionHistory[0].value);
+      setMessageText('please input > or = ' + energyConsumptionHistory.electric[0].value);
     }
   }
+
+  function updateEnergyConsumption2(inputEnergyConsumptionValue) {
```
Comment (stefankorder) [inline src/App.js]: I recommend that you define a function in such a way that you can use it multiple times.

File: src/App.js [hunk 3]
```diff
@@ -1,109 +1,166 @@
...
+      increase: inputEnergyConsumptionValue - energyConsumptionHistory.heating[0].value,
+    };
+    setEnergyConsumptionHistory({
+      electric: [...energyConsumptionHistory.electric],
+      heating: [newInput2, ...energyConsumptionHistory.heating],
+      mobility: [...energyConsumptionHistory.mobility],
+    });
+  }
+
+  function updateEnergyConsumption3(inputEnergyConsumptionValue) {
```
Comment (stefankorder) [inline src/App.js]: I recommend that you define a function in such a way that you can use it multiple times.

File: src/App.js [hunk 4]
```diff
@@ -1,109 +1,166 @@
...
-          borderWidth: '2',
-        },
-      ],
-    };
-    return chartInputData;
-  }
-
-  function updateEnergyConsumption(inputEnergyConsumptionValue) {
-    const newInput = {
+  function updateEnergyConsumption1(inputEnergyConsumptionValue) {
```
Comment (stefankorder) [inline src/App.js]: I recommend that you define a function in such a way that you can use it multiple times.

File: src/components/CategoryButton.js [hunk 5]
```diff
@@ -22,26 +22,29 @@ const CategoryField = styled.section`
...
+  .LinkContainer {
     display: flex;
     justify-content: center;
     align-items: center;
     width: 10vh;
-    height: 100%;
+    height: 10vh;
+    margin: 4px;
     background-color: darkred;
     border-width: 0;
```
Comment (stefankorder) [inline src/components/CategoryButton.js]: Can you please check if removing this CSS property has any effect?

File: src/components/ConfigPersonalButton.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+const ConfigButton = styled.button`
+  width: 50px;
+  height: 50px;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  background-color: white;
+  border-radius: 999px;
```
Comment (stefankorder) [inline src/components/ConfigPersonalButton.js:20]: Do you really need a value of 999px in your case?

File: src/components/Footer.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+  bottom: 0;
+
+  .StyledDashLink {
+    width: 80px;
+    height: 80px;
+    background-color: white;
+    border-style: solid red 3px;
+    display: flex;
+    align-items: center;
+    border-radius: 999px;
```
Comment (stefankorder) [inline src/components/Footer.js]: Do you really need a value of 999px in your case?

File: src/components/Footer.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+const FooterMenu = styled.footer`
+  height: 50px;
+  width: 100%;
+  background-color: black;
+  display: flex;
+  justify-content: space-around;
+  position: absolute;
+  bottom: 0;
+
+  .StyledDashLink {
```
Comment (stefankorder) [inline src/components/Footer.js:30]: It is better to define a separate styled component instead to define a nested class.

File: src/components/Header.js [hunk 9]
```diff
@@ -90,13 +105,33 @@ const Eye = styled.div`
...
+  justify-content: center;
+`;
+
+const MouthLeft = styled.div`
+  height: 3px;
+  width: 28px;
+  transform: rotate(18deg);
+  padding: 1px;
+  background-color: white;
+  border-radius: 1111px;
```
Comment (stefankorder) [inline src/components/Header.js:125]: Do you really need a value of 1111px in your case?

File: src/components/Header.js [hunk 10]
```diff
@@ -90,13 +105,33 @@ const Eye = styled.div`
...
+  top: -6px;
+  left: 3px;
+`;
+const MouthRight = styled.div`
+  height: 3px;
+  width: 25px;
+  transform: rotate(162deg);
   padding: 1px;
   background-color: white;
   border-radius: 1111px;
```
Comment (stefankorder) [inline src/components/Header.js:136]: Do you really need a value of 1111px in your case?

File: src/components/InfoBoardElectric.js [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+export default function InfoBoardMobility({energyConsumptionHistory, dailyTotalBudget, totalConsumption}) {
+  const totalBudget = 30000;
+  const formattedActualDate = new Date().toLocaleDateString('en-GB', {
+    year: 'numeric',
+    month: '2-digit',
+    day: '2-digit',
+  });
+  const saldoIncrease = energyConsumptionHistory.electric[0].increase - dailyTotalBudget;
+  const restBudget = totalBudget - totalConsumption[0];
+  console.log(totalConsumption);
```
Comment (stefankorder) [inline src/components/InfoBoardElectric.js]: Please delete this console log.

File: src/components/InfoBoardMobility.js [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+export default function InfoBoardMobility({energyConsumptionHistory, dailyTotalBudget, totalConsumption}) {
+  const totalBudget = 30000;
+  const formattedActualDate = new Date().toLocaleDateString('en-GB', {
+    year: 'numeric',
+    month: '2-digit',
+    day: '2-digit',
+  });
+  const saldoIncrease = energyConsumptionHistory.mobility[0].increase - dailyTotalBudget;
+  const restBudget = totalBudget - totalConsumption[0];
+  console.log(totalConsumption[1]);
```
Comment (stefankorder) [inline src/components/InfoBoardMobility.js]: Please delete this console log.

**Review by stefankorder — CHANGES_REQUESTED** [review]
Good work! Please take a look at my comments.

File: src/components/CategoryButton.js (hunk 5 above)
Comment (smilenow24) [reply src/components/CategoryButton.js]: Yes, this property is necessary for adjusting the button field - it works fine.

File: src/components/ConfigPersonalButton.js (hunk 6 above)
Comment (smilenow24) [reply src/components/ConfigPersonalButton.js:20]: Yes, for a circle. Belongs to a design construct - it works fine.

File: src/components/Footer.js (hunk 8 above)
Comment (smilenow24) [reply src/components/Footer.js:30]: Ok, I rebuilt it - it works!

File: src/components/Header.js (hunk 9 above)
Comment (smilenow24) [reply src/components/Header.js:125]: Yes, for a circle. Belongs to a design construct - it works fine.

File: src/components/InfoBoardElectric.js (hunk 11 above)
Comment (smilenow24) [reply src/components/InfoBoardElectric.js]: Ok, well observed - have delete it.

File: src/components/InfoBoardMobility.js (hunk 12 above)
Comment (smilenow24) [reply src/components/InfoBoardMobility.js]: Ok.

File: src/App.js (hunk 4 above)
Comment (smilenow24) [reply src/App.js]: Thanks for your Comment: Can you give me a hint how can I solve this in a simple way? - I do not want to lose too much time;

File: src/App.js (hunk 2 above)
Comment (smilenow24) [reply src/App.js]: Thanks for your Comment: Can you give me a hint how can I solve this in a simple way? - I do not want to lose too much time;

File: src/App.js (hunk 3 above)
Comment (smilenow24) [reply src/App.js]: Thanks, a lot for your Review; Can you give me a hint how I can define such functions in best practice?

**Review by stefankorder — APPROVED** [review]
Good work!

**Pull request merged**